lib/adal.js

```diff
@@ -1793,6 +1803,22 @@ var AuthenticationContext = (function () {
         }
     };
 
+    /**
+     * Listens for logouts on other browsing contexts. Clears the user When a logout is detected.
+     * @ignore
+     */
+    AuthenticationContext.prototype._setupCrossBrowsingContextLogoutCleanup = function () {
+        var clearUserOnExternalBrowsingContextLogout = function (evt) {
+            var isExternalBrowsingContextLogout = evt.key === this.CONSTANTS.STORAGE.IDTOKEN && evt.oldValue !== '' && evt.newValue === '';
+
+            if (isExternalBrowsingContextLogout) {
+                this._clearUser();
+            }
+        };
+
+        window.addEventListener('storage', clearUserOnExternalBrowsingContextLogout.bind(this));
```

Comment:
Yep, AFAIK, that's how `storage` listeners must work. Then the handler function is responsible for acting only on the events that are deemed appropriate. 

I considered creating a new storage key for this, but the clearing of id token was pre-existing and does exactly what we want, so I figure it would be okay to use that instead of adding another storage entry and having to clean that up.